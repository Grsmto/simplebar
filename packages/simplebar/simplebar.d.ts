export as namespace SimpleBar;
export = SimpleBar;

declare namespace SimpleBar {
    interface KnownClassNamesOptions {
        contentEl?: string;
        contentWrapper?: string;
        offset?: string;
        mask?: string;
        wrapper?: string;
        placeholder?: string;
        scrollbar?: string;
        track?: string;
        heightAutoObserverWrapperEl?: string;
        heightAutoObserverEl?: string;
        visible?: string;
        horizontal?: string;
        vertical?: string;
        hover?: string;
        dragging?: string;
    }

    type ClassNamesOptions = KnownClassNamesOptions & {
        [className: string]: string;
    };

    interface Options {
        autoHide?: boolean;
        classNames?: ClassNamesOptions;
        forceVisible?: boolean | 'x' | 'y';
        direction?: 'rtl' | 'ltr';
        timeout?: number;
        clickOnTrack?: boolean;
        recalculateThrottleWaitTime?: number;
        scrollbarMinSize?: number;
        scrollbarMaxSize?: number;
    }
}

declare class SimpleBar {
    static removeObserver(): void;
    static instances: Pick<WeakMap<HTMLElement, SimpleBar>, 'get' | 'has'>;

    constructor(element: HTMLElement, options?: SimpleBar.Options);

    recalculate(): void;
    getScrollElement(): HTMLElement;
    getContentElement(): HTMLElement;
    unMount(): void;

    el: HTMLElement;
}
