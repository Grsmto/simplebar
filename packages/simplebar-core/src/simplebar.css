[data-simplebar] {
  position: relative;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-content: flex-start;
  align-items: flex-start;

  --scrollbar-width: 7px;
}

.simplebar-wrapper {
  overflow: hidden;
  width: inherit;
  height: inherit;
  max-width: inherit;
  max-height: inherit;
}

.simplebar-content-wrapper {
  direction: inherit;
  box-sizing: border-box !important;
  position: relative;
  display: block;
  height: 100%; /* Required for horizontal native scrollbar to not appear if parent is taller than natural height */
  width: auto;
  max-width: 100%; /* Not required for horizontal scroll to trigger */
  max-height: 100%; /* Needed for vertical scroll to trigger */
  overflow: auto;
  scrollbar-width: none;
  -ms-overflow-style: none;
}

.simplebar-content-wrapper::-webkit-scrollbar {
  display: none;
  width: 0;
  height: 0;
}

.simplebar-content:before,
.simplebar-content:after {
  content: ' ';
  display: table;
}

.simplebar-placeholder {
  max-height: 100%;
  max-width: 100%;
  width: 100%;
  pointer-events: none;
}

.simplebar-height-auto-observer-wrapper {
  box-sizing: inherit !important;
  height: 100%;
  width: 100%;
  max-width: 1px;
  position: relative;
  float: left;
  max-height: 1px;
  overflow: hidden;
  z-index: -1;
  padding: 0;
  margin: 0;
  pointer-events: none;
  flex-grow: inherit;
  flex-shrink: 0;
  flex-basis: 0;
}

.simplebar-height-auto-observer {
  box-sizing: inherit;
  display: block;
  opacity: 0;
  position: absolute;
  top: 0;
  left: 0;
  height: 1000%;
  width: 1000%;
  min-height: 1px;
  min-width: 1px;
  overflow: hidden;
  pointer-events: none;
  z-index: -1;
}

.simplebar-track {
  z-index: 1;
  position: absolute;
  right: 0;
  bottom: 0;
  pointer-events: none;
  overflow: hidden;
  opacity: 0;
  visibility: hidden;
}

.simplebar-track.simplebar-vertical {
  top: 0;
  width: 11px;
  width: calc(var(--scrollbar-width) + 4px);
}

.simplebar-track.simplebar-horizontal {
  left: 0;
  height: 11px;
}

[data-simplebar].simplebar-dragging {
  pointer-events: none;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -khtml-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

[data-simplebar].simplebar-dragging .simplebar-content {
  pointer-events: none;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -khtml-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

[data-simplebar].simplebar-dragging .simplebar-track {
  pointer-events: all;
}

.simplebar-scrollbar {
  position: absolute;
  left: 0;
  right: 0;
  min-height: 10px;
}

.simplebar-scrollbar:before {
  position: absolute;
  content: '';
  background: black;
  border-radius: 7px;
  left: 2px;
  right: 2px;
  opacity: 0;
  transition: opacity 0.2s 0.5s linear;
}

.simplebar-scrollbar.simplebar-visible:before {
  opacity: 0.5;
  transition-delay: 0s;
  transition-duration: 0s;
}

.simplebar-scrollbar:before {
  top: 2px;
  bottom: 2px;
  left: 2px;
  right: 2px;
}

.simplebar-track.simplebar-horizontal .simplebar-scrollbar {
  right: auto;
  left: 0;
  top: 0;
  bottom: 0;
  min-height: 0;
  min-width: 10px;
  width: auto;
}

.simplebar-scrollable-y .simplebar-track.simplebar-vertical,
.simplebar-scrollable-x .simplebar-track.simplebar-horizontal {
  opacity: 1;
  visibility: visible;
}

/* Rtl support */
[data-simplebar-direction='rtl'] .simplebar-track.simplebar-vertical {
  right: auto;
  left: 0;
}

.simplebar-dummy-scrollbar-size {
  direction: rtl;
  position: fixed;
  opacity: 0;
  visibility: hidden;
  height: 500px;
  width: 500px;
  overflow-y: hidden;
  overflow-x: scroll;
  -ms-overflow-style: scrollbar !important;
}

.simplebar-dummy-scrollbar-size > div {
  width: 200%;
  height: 200%;
  margin: 10px 0;
}
