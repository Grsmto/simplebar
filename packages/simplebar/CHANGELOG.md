# v6.2.1 (Sat Feb 11 2023)

#### üêõ Bug Fix

- Update CHANGELOG.md \[skip ci\] (oi@adriendenat.com)
- Fix typo in README (simplebar/src/simplebar) [#647](https://github.com/Grsmto/simplebar/pull/647) ([@DenisSvistoplasov](https://github.com/DenisSvistoplasov))

#### ‚ö†Ô∏è Pushed to `master`

- feat(core): start adding support for multiple classNames (ref #651) (oi@adriendenat.com)
- fix: fix README (ref #648 #645) (oi@adriendenat.com)
- Publish (oi@adriendenat.com)
- feat(core, vanilla): remove all unnecessary DOM manipulations from core (oi@adriendenat.com)
- Revert "chore: remove unnecessary build step" (oi@adriendenat.com)
- fix(vanilla): fix wrong browser export (fix #639) (oi@adriendenat.com)
- chore: update READMEs (oi@adriendenat.com)

#### Authors: 2

- [@DenisSvistoplasov](https://github.com/DenisSvistoplasov)
- Adrien Denat ([@Grsmto](https://github.com/Grsmto))

---

# v6.2.1 (Sat Feb 11 2023)

#### üêõ Bug Fix

- Fix typo in README (simplebar/src/simplebar) [#647](https://github.com/Grsmto/simplebar/pull/647) ([@DenisSvistoplasov](https://github.com/DenisSvistoplasov))

#### ‚ö†Ô∏è Pushed to `master`

- feat(core): start adding support for multiple classNames (ref #651) ([@Grsmto](https://github.com/Grsmto))
- fix: fix README (ref #648 #645) ([@Grsmto](https://github.com/Grsmto))
- Publish ([@Grsmto](https://github.com/Grsmto))
- feat(core, vanilla): remove all unnecessary DOM manipulations from core ([@Grsmto](https://github.com/Grsmto))
- Revert "chore: remove unnecessary build step" ([@Grsmto](https://github.com/Grsmto))
- fix(vanilla): fix wrong browser export (fix #639) ([@Grsmto](https://github.com/Grsmto))
- chore: update READMEs ([@Grsmto](https://github.com/Grsmto))

#### Authors: 2

- [@DenisSvistoplasov](https://github.com/DenisSvistoplasov)
- Adrien Denat ([@Grsmto](https://github.com/Grsmto))
