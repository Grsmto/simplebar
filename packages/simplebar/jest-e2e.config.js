module.exports = {
    preset: 'jest-puppeteer',
    testRegex: '^.+\\.test\\.e2e\\.js$',
    'transform': {
        "^.+\\.js?$": "babel-jest"
    },
    'transformIgnorePatterns': [
        "[/\\\\]node_modules[/\\\\].+\\.(js|jsx|mjs)$"
    ],
    'moduleFileExtensions': [
        "js"
    ]
};