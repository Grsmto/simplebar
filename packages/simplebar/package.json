{
  "version": "4.2.1",
  "name": "simplebar",
  "title": "SimpleBar.js",
  "description": "Scrollbars, simpler.",
  "files": [
    "dist",
    "src",
    "../../README.md"
  ],
  "author": "Adrien Denat from a fork by Jonathan Nicol",
  "repository": {
    "type": "git",
    "url": "https://github.com/grsmto/simplebar.git",
    "directory": "packages/simplebar"
  },
  "main": "dist/simplebar.min.js",
  "module": "dist/simplebar.esm.js",
  "style": "dist/simplebar.min.css",
  "homepage": "https://grsmto.github.io/simplebar/",
  "bugs": "https://github.com/grsmto/simplebar/issues",
  "license": "MIT",
  "scripts": {
    "start": "webpack-dev-server --mode=development",
    "build": "rollup -c && cp src/simplebar.css dist/simplebar.css && minify dist/simplebar.css > dist/simplebar.min.css && webpack --mode=production",
    "dev": "rollup -c -w",
    "test": "yarn test:unit && yarn test:e2e",
    "test:unit": "jest -c jest-unit.config.js",
    "test:e2e": "jest --runInBand -c jest-e2e.config.js",
    "version": "yarn build"
  },
  "dependencies": {
    "can-use-dom": "^0.1.0",
    "core-js": "^3.0.1",
    "lodash.debounce": "^4.0.8",
    "lodash.memoize": "^4.1.2",
    "lodash.throttle": "^4.1.1",
    "resize-observer-polyfill": "^1.5.1"
  },
  "devDependencies": {
    "css-loader": "^0.28.11",
    "minify": "^3.0.5",
    "promise": "^8.0.2",
    "react-select": "^2.4.3",
    "style-loader": "^0.21.0"
  }
}
