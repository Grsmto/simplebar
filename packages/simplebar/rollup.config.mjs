import { createRequire } from 'node:module';
import resolve from '@rollup/plugin-node-resolve';
import commonjs from '@rollup/plugin-commonjs';
import typescript from '@rollup/plugin-typescript';
import license from 'rollup-plugin-license';
import { terser } from 'rollup-plugin-terser';
import { getBanner, getExternals } from '../../rollup.config.mjs';

const require = createRequire(import.meta.url);
const pkg = require('./package.json');

const builds = [
  // ES module (for bundlers) build.
  {
    input: 'src/index.ts',
    external: getExternals(pkg),
    output: {
      file: pkg.module,
      format: 'esm',
      sourcemap: true,
    },
    plugins: [
      typescript({ tsconfig: '../../tsconfig.json' }),
      license(getBanner(pkg)),
    ],
  },
];

if (process.env.BUILD !== 'development') {
  // UMD build
  builds.push({
    input: 'src/index.ts',
    external: getExternals(pkg),
    output: {
      name: 'SimpleBar',
      file: pkg.main,
      format: 'umd',
      exports: 'named',
      globals: {
        'simplebar-core': 'SimpleBar',
      },
    },
    plugins: [
      typescript({ tsconfig: '../../tsconfig.json' }),
      license(getBanner(pkg)),
    ],
  });

  // browser script tag build, minified
  builds.push({
    input: 'src/index.ts',
    output: {
      name: 'SimpleBar',
      file: pkg.unpkg,
      format: 'iife',
    },
    plugins: [
      resolve(),
      commonjs(),
      typescript({ tsconfig: '../../tsconfig.json' }),
      terser(),
      license(getBanner(pkg)),
    ],
  });

  builds.push(
    // browser script tag build, non-minified
    {
      input: 'src/index.ts',
      output: {
        name: 'SimpleBar',
        file: 'dist/simplebar.js',
        format: 'iife',
      },
      plugins: [
        resolve(),
        commonjs(),
        typescript({ tsconfig: '../../tsconfig.json' }),
        license(getBanner(pkg)),
      ],
    }
  );
}

export default builds;
