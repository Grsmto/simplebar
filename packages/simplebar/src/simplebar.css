[data-simplebar] {
    position: relative;
    z-index: 0;
    overflow: hidden!important;
    max-height: inherit;
    -webkit-overflow-scrolling: touch; /* Trigger native scrolling for mobile, if not supported, plugin is used. */
}

[data-simplebar="init"] {
    display: flex;
}

[data-simplebar] .simplebar-scroll-content,
[data-simplebar] .simplebar-content {
    overflow: hidden;
}

[data-simplebar="init"] .simplebar-scroll-content,
[data-simplebar="init"] .simplebar-content {
    overflow: scroll;
}

.simplebar-scroll-content {
    overflow-x: hidden!important;
    min-width: 100%!important;
    max-height: inherit!important;
    box-sizing: content-box!important;
}

.simplebar-content {
    overflow-y: hidden!important;
    box-sizing: border-box!important;
    min-height: 100%!important;
}

.simplebar-track {
    z-index: 1;
    position: absolute;
    right: 0;
    bottom: 0;
    width: 11px;
    pointer-events: none;
}

.simplebar-scrollbar {
    position: absolute;
    right: 2px;
    width: 7px;
    min-height: 10px;
}

.simplebar-scrollbar:before {
    position: absolute;
    content: "";
    background: black;
    border-radius: 7px;
    left: 0;
    right: 0;
    opacity: 0;
    transition: opacity 0.2s linear;
}

.simplebar-track:hover .simplebar-scrollbar:before,
.simplebar-track .simplebar-scrollbar.visible:before {
    /* When hovered, remove all transitions from drag handle */
    opacity: 0.5;
    transition: opacity 0 linear;
}

.simplebar-track.vertical {
    top: 0;
}

.simplebar-track.vertical.rtl {
    right: unset;
    left: 0;
}

.simplebar-track.vertical .simplebar-scrollbar:before {
    top: 2px;
    bottom: 2px;
}

.simplebar-track.horizontal {
    left: 0;
    width: auto;
    height: 11px;
}

.simplebar-track.horizontal .simplebar-scrollbar:before {
    height: 100%;
    left: 2px;
    right: 2px;
}

.horizontal.simplebar-track .simplebar-scrollbar {
    right: auto;
    left: 0;
    top: 2px;
    height: 7px;
    min-height: 0;
    min-width: 10px;
    width: auto;
}

/* Rtl support */
[data-simplebar-direction="rtl"] .simplebar-track {
    right: auto;
    left: 0;
}

[data-simplebar-direction="rtl"] .simplebar-track.horizontal {
    right: 0;
}
