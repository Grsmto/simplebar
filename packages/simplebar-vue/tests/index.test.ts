import { shallowMount, destroyWrapper } from './test-utils';
import simplebar from '../component';
import SimpleBar from 'simplebar-core';

describe('simplebar', () => {
  describe('snapshots', () => {
    /**
     * jest-serializer-vue is not compatible with vue3 yet, so you
     * cannot pass the wrapper object directly. Call .html() before
     * @see {@link https://github.com/eddyerburgh/jest-serializer-vue/pull/56}
     */
    it('renders without crashing', () => {
      const wrapper = shallowMount(simplebar);
      expect(wrapper.html()).toMatchSnapshot();
      destroyWrapper(wrapper);
    });

    it('renders with default slot', () => {
      const wrapper = shallowMount(simplebar, {
        slots: {
          default: '<div class="inner-content" />',
        },
      });
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  it('can access SimpleBar instance', () => {
    const wrapper = shallowMount(simplebar);
    expect(wrapper.vm.SimpleBar).toBeInstanceOf(SimpleBar);
  });

  it('can access root element ref property', () => {
    const wrapper = shallowMount(simplebar);
    expect(wrapper.element).toEqual(wrapper.vm.$refs.element);
  });

  it('can access scrollElement property', () => {
    const wrapper = shallowMount(simplebar);
    const scrollElement = wrapper.vm.scrollElement;

    expect(scrollElement).toEqual(
      wrapper.find('.simplebar-content-wrapper').element
    );
  });

  it('can access contentElement property', () => {
    const wrapper = shallowMount(simplebar);
    const scrollElement = wrapper.vm.contentElement;

    expect(scrollElement).toEqual(wrapper.find('.simplebar-content').element);
  });

  it('works with options as  data attributes', () => {
    const wrapper = shallowMount(simplebar, {
      attrs: { 'data-simplebar-force-visible': 'true', 'tabIndex': -1 },
    });
    expect(wrapper.vm.SimpleBar.options.forceVisible).toEqual(true);
    expect(wrapper.vm.SimpleBar.options.tabIndex).toEqual(-1);
  });

  it('works with options as props', () => {
    const wrapper = shallowMount(simplebar, {
      propsData: { forceVisible: true },
    });
    expect(wrapper.vm.SimpleBar.options.forceVisible).toEqual(true);
  });

  it('emits a scroll event', async () => {
    const wrapper = shallowMount(simplebar);
    const scrollElement = wrapper.find('.simplebar-content-wrapper');

    expect(wrapper.emitted()).not.toHaveProperty('scroll');
    await scrollElement.trigger('scroll');
    expect(wrapper.emitted()).toHaveProperty('scroll');
  });

  it('destroys Simplebar instance when component is unmounted to prevent memory leaks', () => {
    // expect.assertions(1);
    const wrapper = shallowMount(simplebar);
    const instance = wrapper.vm.SimpleBar;
    jest.spyOn(instance, 'unMount');

    destroyWrapper(wrapper);
    expect(instance.unMount).toHaveBeenCalledTimes(1);
  });
});
