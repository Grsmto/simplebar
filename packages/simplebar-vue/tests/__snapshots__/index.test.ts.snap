// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simplebar snapshots renders with default slot 1`] = `
<div data-simplebar="init">
  <div class="simplebar-wrapper">
    <div class="simplebar-content-wrapper" tabindex="0" role="region" aria-label="scrollable content">
      <div class="simplebar-content">
        <div class="inner-content"></div>
      </div>
    </div>
    <div class="simplebar-track simplebar-horizontal">
      <div class="simplebar-scrollbar" style="width: 0px; display: none;"></div>
    </div>
    <div class="simplebar-track simplebar-vertical">
      <div class="simplebar-scrollbar" style="height: 0px; display: none;"></div>
    </div>
  </div>
  <div class="simplebar-observer">
    <div class="simplebar-observer-inner"></div>
  </div>
</div>
`;

exports[`simplebar snapshots renders without crashing 1`] = `
<div data-simplebar="init">
  <div class="simplebar-wrapper">
    <div class="simplebar-content-wrapper" tabindex="0" role="region" aria-label="scrollable content">
      <div class="simplebar-content"></div>
    </div>
    <div class="simplebar-track simplebar-horizontal">
      <div class="simplebar-scrollbar" style="width: 0px; display: none;"></div>
    </div>
    <div class="simplebar-track simplebar-vertical">
      <div class="simplebar-scrollbar" style="height: 0px; display: none;"></div>
    </div>
  </div>
  <div class="simplebar-observer">
    <div class="simplebar-observer-inner"></div>
  </div>
</div>
`;
